vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+
+### Binary package ABI
+
+The full Application Binary Interface (ABI) of the binary package is in the debug output. For example, for zlib:
+
+```
+[DEBUG] Trying to hash <path>\buildtrees\zlib\x86-windows.vcpkg_abi_info.txt
+[DEBUG] <path>\buildtrees\zlib\x86-windows.vcpkg_abi_info.txt has hash bb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87
+```
+
+The ABI hash `bb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87` for package zlib is constructed by hashing all the possible relevant information (sources, compiler information) to distinguish binary packages. The binary cache provider uses this as a key to refer to the binary package artifact. If this hash were to change between runs for the same package, it would mean some part of the package had changed.
+
+### Creating the binary package
+
+A successful creation of the binary package should output:
+```
+[DEBUG] 1010: CreateProcessW("<path>\downloads\tools\nuget-6.2.1-windows\nuget.exe" pack -ForceEnglishOutput -Verbosity detailed -NonInteractive "<path>\buildtrees\zlib\x86-windows.nuspec" -OutputDirectory "<path>\buildtrees" -NoDefaultExcludes)
+[DEBUG] 1010: cmd_execute_and_stream_data() returned 0 after  1038094 us
+...
+Successfully created package '<path>\buildtrees\zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg'.
+```
+If you are not seeing this message, consider the following:
+
+* Make sure `CreateProcessW` returns a successful exit code `0` when compressing the package into an artifact
+* Make sure you have enough disk space on your machine
+
+### Failing to write to the binary cache
+
+A successful write to the binary cache should have the following output for each dependency:
+```
+[DEBUG] 1011: CreateProcessW("<path>\downloads\tools\nuget-6.2.1-windows\nuget.exe" push -ForceEnglishOutput -Verbosity detailed -NonInteractive "<path>\buildtrees\zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg" -Timeout 100 -Source <NuGet Feed>)
+[DEBUG] 1011: cmd_execute_and_stream_data() returned 0 after   767027 us
+NuGet Version: 6.2.1.2
+Pushing zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg to '<NuGet Feed>'...
+Your package was pushed.
+Stored binaries in 1 destinations in 1.8 s.
+```
+
+Your NuGet feed or cache should look something like this:
+```
+Mode                 LastWriteTime         Length Name
+----                 -------------         ------ ----
+-a---           8/16/2023  8:53 PM         529044 zlib_x86-windows.1.2.13-vcpkgbb1c96759ac96102b4b18215db138daedd3eb16c2cd3302ae7bffab2b643eb87.nupkg
+```
+
+If writing to the cache is failing, consider the following:
+
+* Make sure your binary source has `write` permissions
```

Comment:
This wording is potentially confusing -- at the vcpkg configuration level, `readwrite` isn't a permission.

```suggestion
* Make sure your [binary cache configuration](binarycaching.md#configuration-syntax) is set to `write` or `readwrite`
```